src/includes/class_building.php

```diff
@@ -89,23 +89,26 @@ public function calendarList() {
 		$buildings = $this->getall();
 
 		$output = "<ul>";
-		foreach ($buildings as $building) {
+		
+    if($buildings) {
```

Comment:
This if statement should be removed. Instead, ensure that $buildings is always an array. Buildings is assigned on line 89, with the getall() method.  

Look at changing getall() to return an array when there are no buildings. It can still set the error message as well.